app/models/service_template.rb

```diff
@@ -45,6 +60,22 @@ class ServiceTemplate < ApplicationRecord
   virtual_has_one :custom_actions, :class_name => "Hash"
   virtual_has_one :custom_action_buttons, :class_name => "Array"
 
+  def self.create_catalog_item(options, auth_user)
+    options[:config_info].assert_valid_keys(CATALOG_ITEM_ATTRS)
```

Comment:
I am not convinced we want to validate keys this way which makes maintaining the list difficult. I suggest we not limit keys here and leave the backend further down to raise error if invalid keys are encountered.

I like your way to create a template `create(options.except(:config_info))`. This way we can accept for example either `options[:service_template_catalog_id] = 1` or `options[:service_template_catalog]=catalog_obj`.